src/Microsoft.PowerApps.TestEngine/System/FileSystem.cs

```diff
@@ -1,91 +1,230 @@
+            }
+            else
+            {
+                throw new InvalidOperationException(string.Format("Invalid file path {0}.", fileName));
+            }
+        }
+
+        public string[] GetFiles(string directoryName)
+        {
+            if (IsNonUNCDirectoryPath(directoryName))
+            {
+                return Directory.GetFiles(directoryName);
+            }
+            else
+            {
+                throw new InvalidOperationException(string.Format("Path invalid or read from path: {0} not permitted.", directoryName));
+            }
+        }
+
+        public string[] GetFiles(string directoryName, string searchPattern)
+        {
+            if (IsNonUNCDirectoryPath(directoryName))
+            {
+                return Directory.GetFiles(directoryName, searchPattern);
+            }
+            else
+            {
+                throw new InvalidOperationException(string.Format("Path invalid or read from path: {0} not permitted.", directoryName));
+            }
+        }
+
+        public void WriteTextToFile(string filePath, string text)
+        {
+            if (IsWritePermittedFilePath(filePath))
+            {
+                if (File.Exists(filePath))
+                {
+                    File.AppendAllText(filePath, text);
+                }
+                else
+                {
+                    File.WriteAllText(filePath, text);
+                }
+            }
+            else
+            {
+                throw new InvalidOperationException(string.Format("Write to path: {0} not permitted, ensure path is rooted in base TestEngine path.", filePath));
+            }
+        }
+
+        public void WriteTextToFile(string filePath, string[] text)
+        {
+            if (IsWritePermittedFilePath(filePath))
+            {
+                if (File.Exists(filePath))
+                {
+                    File.AppendAllLines(filePath, text);
+                }
+                else
+                {
+                    File.WriteAllLines(filePath, text);
+                }
+            }
+            else
+            {
+                throw new InvalidOperationException(string.Format("Write to path: {0} not permitted, ensure path is rooted in base TestEngine path.", filePath));
+            }
+        }
+
+        public bool IsValidFilePath(string filePath)
+        {
+            try
+            {
+                if (string.IsNullOrWhiteSpace(filePath))
+                {
+                    return false;
+                }
+                var fullPath = Path.GetFullPath(filePath);
+
+                //check if its a network path if so fail
+                var fullPathUri = new Uri(fullPath.StartsWith(@"\\?\") ? fullPath.Replace(@"\\?\", "") : fullPath);
+                if (fullPathUri.IsUnc)
+                {
+                    return false;
+                }
+                //TODO: additional checks for reserved folder names

```

Comment:
Is this something we are planning to do in the future? How important it is? 
Usually, TODO comments are hard to track and revisit, and I would suggest instead if this is an important functionality to create a work item. If is something simple, then let's implement now.

Reply:
added